resources/js/Pages/Home.vue

```diff
@@ -42,18 +43,51 @@
             user: Object,
         },
         methods: {
+            checkEmpty() {
+                if( !this.form.itemsInput ) {
+                    this.error = {
+                        type: 'warning',
+                        message: 'item-form-error-message-empty'
+                    };
+                    return false;
+                }
+                return true;
+            },
+            validate() {
+                let valid = true;
+                const lines = this.form.itemsInput.split( '\n' );
+                for ( let line of lines ) {
+                    line = line.trim();
+                    valid = valid && ( line == '' || line.match( /^Q[0-9]*$/ ) );
+                }
+
+                if( !valid ) {
+                    this.error = {
+                        type: 'error',
+                        message: 'item-form-error-message-invalid'
+                    };
+                    return false;
+                }
+
+                return lines.join( '|' );
+            },
             send() {
-                // TODO: Implement validation
+                if( !this.checkEmpty() ) {
+                    return;
+                }
 
-                const idsToSend = this.form.itemsInput.split('\n').join('|');
-                this.$inertia.get('/results?ids=' + idsToSend)
+                const validatedIds = this.validate();
+                if( validatedIds ) {
+                    this.$inertia.get( '/results?ids=' + validatedIds );
+                }
```

Comment:
> Since we are already mutating the internal state of the component...

To build up on that idea, if we were just to check the error prop in this function it would reduce the b`if branching here to basically one, something like this:

```js
send() {
    this.validate();

    if (this.error) {
        return;
    }
   
    this.$inertia.get( '/results?ids=' + validatedIds );
}

Reply:
Nice, I like this. I also tried the computed properties, but it looks like they are not really needed here.